services/workflows-service/scripts/alerts/generate-alerts.ts

```diff
@@ -575,9 +577,13 @@ export const getAlertDefinitionCreateData = (
     config: { config: {} },
     tags: [],
     additionalInfo: {},
-    ...extraColumns,
-    projectId: project.id,
-  };
+    crossEnvKey: extraColumns.crossEnvKey || 'test' + Math.random(),
```

Comment:
in case this is not optional field anymore you can remove the default value...
```suggestion
    crossEnvKey: extraColumns.crossEnvKey
```

Reply:
oh forgot about that, the issue was that it generate the same envKey and failed to insert - do you know why?